### PR TITLE
Mailgun #191 Email Validation

### PR DESCRIPTION
**Pull request by DavidGarciaCat**
Validates an Email Address https://github.com/mailgun/mailgun-php/issues/191

ToDo:
- Confirm with @Nyholm about the best way to proceed with the API keys and implement it.
- Make sure the PHP INTL extension won't be a problem for Spoof Checks https://github.com/mailgun/mailgun-php/issues/191#issuecomment-369764208

### CODE REVIEW

**Comment by Nyholm** [comment]
Sorry for the super late reply. 
I know we discussed this offline before, anyhow, I'll write this down just for the record. 

I like this PR. But I dont think the API client should validate emails itself. It should just pass validation calls to the server.

File: composer.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
         "php-http/message": "^1.0",
         "php-http/client-common": "^1.1",
         "php-http/discovery": "^1.0",
-        "webmozart/assert": "^1.2"
+        "webmozart/assert": "^1.2",
+        "egulias/email-validator": "^2.1"
```
Comment (Nyholm) [inline composer.json]: This should be removed.

Comment (DavidGarciaCat) [reply composer.json]: Yep, agree. I’ll proceed with the steps that we discussed the last time we met and discussed this case. I’ll keep you posted for a final review.

**Comment by Nyholm** [comment]
Fixed the merged conflict. =)

**Comment by DavidGarciaCat** [comment]
Hey @Nyholm 

I've needed to rebase due to a conflict didn't allow to merge; we can merge the PR once the new Travis Built is passing.

I think that more testing is needed before to release v3, especially due to the recent updates (we have updated a lot of files0; I wouldn't like to release v3 and in a couple of days push a bug fix.

Cheers,

**Comment by Nyholm** [comment]
Wohoo!

**Pull request merged**